source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -504,7 +503,9 @@ Http::FilterHeadersStatus JsonTranscoderFilter::encodeHeaders(Http::ResponseHead
   }
 
   headers.setReferenceContentType(Http::Headers::get().ContentTypeValues.Json);
-  if (!method_->descriptor_->server_streaming()) {
+
+  if (!method_->descriptor_->server_streaming() ||
+      (method_->descriptor_->server_streaming() && method_->response_type_is_http_body_)) {
```

Comment:
logically, you don't need the 2nd server_streaming check

Reply:
Thanks for catch, simplified.